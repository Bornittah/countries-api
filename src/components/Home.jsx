import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Header from './shared/Header/Header';
import HomeScreen from './shared/HomeScreen/HomeScreen';
import Grid from './shared/Grid';
import { fetchCountries } from '../redux/countries';

const Home = () => {
  const [loading, setLoading] = useState(false);
  const countries = useSelector((state) => state.entities.countries);
  const ctrs = [];
  countries.map((data) => {
    ctrs.push(data.region);
    return ctrs;
  });

  const regions = [...new Set(ctrs)];
  const countCountries = (array, region) => array.filter((item) => item === region).length;
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCountries());
    setLoading(true);
  }, []);

  return (
    <div>
      <Header title="The World" />
      <HomeScreen title="World" count={regions.length} units="Major Regions" />
      <h2 className="title">STAT BY REGIONS</h2>
      <div className="grid-container">
        {loading ? (
          regions.map((region) => (
            <Grid key={region} title={region} number={countCountries(ctrs, region)} />
          ))) : ('loading data...')}
      </div>
    </div>
  );
};

export default Home;
