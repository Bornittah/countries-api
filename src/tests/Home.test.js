import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Home from '../components/Home';

describe('Testing Home component', () => {
  it('Testing with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
