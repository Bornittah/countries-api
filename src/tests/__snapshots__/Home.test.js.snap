// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home component Testing with snapshot 1`] = `
<div>
  <header
    className="header"
  >
    <nav>
      <ul
        className="menu flex space-btn"
      >
        <li>
          <a
            className="link"
            href="/"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
              />
            </svg>
          </a>
        </li>
        <li>
          Home
        </li>
        <li>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="home-screen"
  >
    <h2>
      The World
    </h2>
    <ul
      className="flex space-btn home-data"
    >
      <li
        className="flex flex-column text-right"
      >
        <span
          className="data-number"
        >
          0
        </span>
        <span>
          Major Regions
        </span>
      </li>
      <li
        className="flex flex-column text-right"
      >
        <span
          className="data-number"
        >
          0
        </span>
        <span>
          Countries
        </span>
      </li>
    </ul>
  </div>
  <div
    className="grid-container"
  >
    loading data...
  </div>
</div>
`;
